src/pybullet_industrial/gcode_class.py

```diff
@@ -0,0 +1,455 @@
+            if not np.isnan(g_com):
+                self.last_point = np.array(self.new_point)
+                self.last_or = np.array(self.new_or)
+
+                # Checking for interpolation commands
+                if g_com in [0, 1, 2, 3]:
+
+                    # Setting the new point considering offset
+                    self.new_point = np.array([0.0, 0.0, 0.0])
+                    c = 0
+
+                    for n in xyz_val:
+
+                        if np.isnan(n):
+                            self.new_point[c] = self.last_point[c]
+
+                        else:
+                            self.new_point[c] = n + self.offset[0][c]
+
+                        c += 1
+
+                    # Setting the new orientation considering offset
+                    self.new_or = np.array([0.0, 0.0, 0.0])
+                    c = 0
+
+                    for n in abc_val:
+
+                        if np.isnan(n):
+                            self.new_or[c] = self.last_or[c]
+
+                        else:
+                            self.new_or[c] = n + self.offset[1][c]
+
+                        c += 1
+
+                    # Moving the endeffector if there is a G0 Interpolation
+                    if g_com == 0:
+                        self.g0_interpolation()
+
+                    # Moving the endeffector if ther is a G1 Interpolation
+                    elif g_com in [1, 2, 3]:
+                        self.g123_interpolation(g_com, r_val)
+
+                # Activation of the zero offset
+                elif g_com == 54:
+                    self.offset = np.array([self.last_point, self.last_or])
+
+                # Deactivation of the zero offset
+                elif g_com == 500:
+                    self.offset = np.array([[0.0, 0.0, 0.0],
+                                           [0.0, 0.0, 0.0]])
+
+                # Plane selelection for circular interpolation
+                elif g_com == 17:
+                    self.plane = 2  # X-Y Plane
+
+                elif g_com == 18:
+                    self.plane = 1  # X-Z Plane
+
+                elif g_com == 19:
+                    self.plane = 0  # Y-Z Plane
+
+            # Checking for a M-command
+            elif not np.isnan(m_com):
+                self.m_commands[m_com][0]()
+
+            # Checking for a T-command
+            elif not np.isnan(t_com):
+                self.t_commands[t_com][0]()
+                self.calibrate_tool()
+
+    def g0_interpolation(self):
+        """This method is used in the run_gcode() Method. The G0-interpolation
+        sets the position of the endeffector or tool without running an
+        Interpolation command
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        orientation = p.getQuaternionFromEuler(self.new_or)
+        e = self.active_endeffector
+
+        if not self.active_endeffector == -1:
+
+            for _ in range(15):
```

Comment:
It would probably be best if the run_gcode function returns a iterator that performs the loop. This would make everything much more modular and usable